packages/core/src/api/monographs.ts

```diff
@@ -109,13 +96,17 @@ export class Monographs {
       false
     );
 
-    const monograph: Monograph = {
+    const userKey = await this.db.user.getEncryptionKey();
+    const monograph: MonographApiRequest = {
       id: noteId,
       title: note.title,
       userId: user.id,
       selfDestruct: opts.selfDestruct || false,
       ...(opts.password
         ? {
+            password: userKey
+              ? await this.db.storage().encrypt(userKey, opts.password)
+              : undefined,
```

Comment:
Now that I think about it, this is also a bad idea. We'll have to re-encrypt and re-publish all these passwords whenever user's password changes.

What if instead of using user's encryption key, we create a new sub-master key just like we do in the case of attachments, and then use that to encrypt the password?